### PR TITLE
Rewrite "SSL Setup" documentation

### LINKED ISSUE #60: Improve SSL setup / documentation
It looks like quite a few have problems getting the SSL setup done right:

image

We could improve documentation but maybe a separate setup ABAP would be helpful that checks parameters and uploads/updates the required certificates. 

https://docs.abapgit.org/guide-ssl-setup.html

https://docs.abapgit.org/other-test-ssl.html

### PR DESCRIPTION
**Pull request by mbtools**
Complete rewrite

Closes #60

### CODE REVIEW

File: docs/guide-ssl-setup.md [hunk 1]
```diff
@@ -4,67 +4,150 @@ category: setup
...
+ssl/client_ciphersuites      = 150:PFS:HIGH::EC_X25519:EC_P256:EC_HIGH
+icm/HTTPS/client_sni_enabled = TRUE
+ssl/client_sni_enabled       = TRUE
+
+SETENV_26 = SECUDIR=$(DIR_INSTANCE)$(DIR_SEP)sec
+SETENV_27 = SAPSSL_CLIENT_CIPHERSUITES=150:PFS:HIGH::EC_X25519:EC_P256:EC_HIGH
+SETENV_28 = SAPSSL_CLIENT_SNI_ENABLED=TRUE 
+```
+
+To add profile parameters use transaction `RZ10` or update contents of file `DEFAULT.PFL` directly. A system restart might be required.
```
Comment (nesl9) [inline docs/guide-ssl-setup.md:51]: add comma -> parameters, use transaction 

add "the" -> update the contents

File: docs/guide-ssl-setup.md [hunk 2]
```diff
@@ -4,67 +4,150 @@ category: setup
...
+
+If configured correctly, transaction `SMICM` > Goto > Services (`Shift + F1`) will show a green checkmark next to the HTTPS service.
+
+![ssl_setup_services](img/ssl_setup_services.png)
+
+### SAP Trust Manager
+
+As a default, abapGit is using an anonymous client connection (`ANONYM - SSL Client (Anonymous)`). A secure connection requires that the Git server certificate is available in the certificate list. The certificate must also be valid!
+
+First, download the server certifates to you local machine. Then import and add these certificates to your SAP system.
```
Comment (nesl9) [inline docs/guide-ssl-setup.md:66]: typos -> certificates to your

File: docs/guide-ssl-setup.md [hunk 3]
```diff
@@ -4,67 +4,150 @@ category: setup
...
+- "A4H - SAP NetWeaver AS ABAP 7.4 and above (Linux / SAP HANA)" - `/usr/sap/A4H/sys/profile/DEFAULT.PFL`
+- "NPL - SAP NetWeaver 7.x" - `/sapmnt/NPL/profile/DEFAULT.PFL`
+
+If configured correctly, transaction `SMICM` > Goto > Services (`Shift + F1`) will show a green checkmark next to the HTTPS service.
+
+![ssl_setup_services](img/ssl_setup_services.png)
+
+### SAP Trust Manager
+
+As a default, abapGit is using an anonymous client connection (`ANONYM - SSL Client (Anonymous)`). A secure connection requires that the Git server certificate is available in the certificate list. The certificate must also be valid!
```
Comment (nesl9) [inline docs/guide-ssl-setup.md:64]: change tense -> abapGit uses an

File: docs/guide-ssl-setup.md [hunk 4]
```diff
@@ -4,67 +4,150 @@ category: setup
...
+If done correctly, transaction `STRUST` will show the Git server certificates in the certificate list.
+
+![ssl_setup_trust](img/ssl_setup_trust.png)
+
+#### Download the Certificate Files
+
+##### Option A - Chrome 
+
+1. Using Google Chrome to go to [https://github.com](https://github.com)
+2. Click on the lock icon near the address bar, then click on "Connection is sercure"
```
Comment (nesl9) [inline docs/guide-ssl-setup.md:79]: typo -> secure

File: docs/guide-ssl-setup.md [hunk 5]
```diff
@@ -4,67 +4,150 @@ category: setup
...
+2. Click on the Change button
+3. Open "SSL client Client SSL Client (Anonymous)" folder
+4. In the third box called "Certificate", click on the bottom-left button "Import certificate" to bring the certificate into the system
+5. Select "Add to certificate list"
+6. Repeat the process for all downloaded certificates
+7. Save
+
+### abapGit Settings
+
+If your SAP system is behind a firewall, it might require a HTTP proxy to access the Git server. If this is the case, maintain the proxy host, port, and authentication (user/password) in your global abapGit settings.
```
Comment (nesl9) [inline docs/guide-ssl-setup.md:118]: change article -> an HTTP

File: docs/guide-ssl-setup.md [hunk 6]
```diff
@@ -4,67 +4,150 @@ category: setup
...
-![](img/strust1.png)
+### Troubleshooting
 
-Report [ZABAPGIT_TEST_SSL](./other-test-ssl.html) can be used to verify that the connection works.
+1. Go to transaction `SMICM`
+2. Check that ICM is in status "running" (green light)
+3. Select Goto > Services and check that the HTTPS service is active (green checkmark)
+4. Select Goto > Trace Files > Display All to view the ICM trace (`dev_icm`)
+- Check for any errors
+- Find the sections called "SSL Initialization" and check if it ends with "Success - SapCryptoLib SSL ready!"
```
Comment (nesl9) [inline docs/guide-ssl-setup.md:133]: singular -> Find the section

File: docs/guide-ssl-setup.md [hunk 7]
```diff
@@ -4,67 +4,150 @@ category: setup
...
-ssl/client_ciphersuites = 150:PFS:HIGH::EC_P256:EC_HIGH
-ssl/ciphersuites = 135:PFS:HIGH::EC_P256:EC_HIGH
-icm/HTTPS/client_sni_enabled = TRUE
-```
+#### Accessing GitHub
 
-To add profile parameters use transaction RZ10 or update contents of file DEFAULT.PFL directly. For [preview, evaluation and developer versions of SAP NetWeaver](https://go.support.sap.com/minisap/#/minisap) refer to following locations, depending on your system name:
-- "A4H - SAP NetWeaver AS ABAP 7.4 and above (Linux / SAP HANA)" - /usr/sap/A4H/sys/profile/DEFAULT.PFL
-- "NPL - SAP NetWeaver 7.x" - /sapmnt/NPL/profile/DEFAULT.PFL
+On [2018-02-22](https://githubengineering.com/crypto-removal-notice/), GitHub deprectated weak TLS connections.  See [SAP Note 510007](https://launchpad.support.sap.com/#/notes/510007) for details on required profile parameter changes.
```
Comment (nesl9) [inline docs/guide-ssl-setup.md:153]: type -> deprecated

**Review by nesl9 — CHANGES_REQUESTED** [review]
Thanks @mbtools! The changes look great and I love the gif. 

I found some minor changes/typos.

**Comment by mbtools** [comment]
Grammarly is working for issues , PRs, and comments but I wish it was working in the GH file editor as well 😉 I clearly need it 

Thanks for checking, @nesl9

**Comment by AndreaBorgia-Abo** [comment]
Very nice, thank you both

**Pull request merged**